homeassistant/components/sensor/google_travel_time.py

```diff
@@ -78,52 +84,70 @@ def convert_time_to_utc(timestr):
 def setup_platform(hass, config, add_devices_callback, discovery_info=None):
     """Setup the travel time platform."""
     # pylint: disable=too-many-locals
-    options = config.get(CONF_OPTIONS)
+    def run_setup(event):
+        """Delay the setup until home assistant is fully initialized.
+
+        This allows any entities to be created already
+        """
+        options = config.get(CONF_OPTIONS)
 
-    if options.get('units') is None:
-        if hass.config.temperature_unit is TEMP_CELSIUS:
-            options['units'] = 'metric'
-        elif hass.config.temperature_unit is TEMP_FAHRENHEIT:
-            options['units'] = 'imperial'
+        if options.get('units') is None:
+            if hass.config.temperature_unit is TEMP_CELSIUS:
+                options['units'] = 'metric'
+            elif hass.config.temperature_unit is TEMP_FAHRENHEIT:
```

Comment:
I can change this. But I wanted to avoid changing the existing behavior too much. This just pushed into another method to hold of setup until homeassistant is ready.